share/functions/ls.fish

```diff
@@ -50,7 +50,6 @@ function ls --description "List contents of directory"
     __fish_set_lscolors
 
     isatty stdout
```

Comment:
The intention is to apply -F only if the output goes to the terminal.

The simplest fix is to set $opt before:

```fish
set -l opt
isatty stdout
and set opt -F
```

Note that we have some cases above where we might apply "-F" even if the output isn't a tty, those should be removed. It's either when we have `colorls` or when `ls` doesn't support any of the color flags.